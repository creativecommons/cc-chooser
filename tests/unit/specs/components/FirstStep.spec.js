import { createLocalVue, mount } from '@vue/test-utils'
import Buefy from 'buefy'
import FirstStep from '@/components/FirstStep.vue'

const localVue = createLocalVue()

localVue.use(Buefy)

describe('FirstStep Component', () => {
    let wrapper

    beforeEach(() => {
        wrapper = mount(FirstStep, {
            localVue,
            mocks: {
                $t: key => key
            }
        })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('Mark up is correctly rendered', () => {
        wrapper.setProps({
            selected: undefined,
            stepId: 0,
            status: 'current'
        })

        expect(wrapper.element).toMatchSnapshot()
    })

    it('Radio Input value is YES', () => {
        wrapper.setProps({
            selected: undefined,
            stepId: 0,
            status: 'current'
        })

        const radio = wrapper.findAll('input[type="radio"]').at(0)
        radio.setChecked()

        expect(wrapper.emitted().change[0]).toEqual(['FS', 0, true])
    })

    it('Radio Input value is NO', () => {
        wrapper.setProps({
            selected: undefined,
            stepId: 0,
            status: 'current'
        })

        const radio = wrapper.findAll('input[type="radio"]').at(1)
        radio.setChecked()

        expect(wrapper.emitted().change[0]).toEqual(['FS', 0, false])
    })

    it('props:selected false', () => {
        wrapper.setProps({
            selected: false,
            stepId: 0,
            status: 'current'
        })

        expect(wrapper.vm.radio).toBe('no')
        expect(wrapper.vm.cardText).toBe('stepper.FS.not-selected')
    })

    it('props:selected true', () => {
        wrapper.setProps({
            selected: true,
            stepId: 0,
            status: 'current'
        })

        expect(wrapper.vm.radio).toBe('yes')
        expect(wrapper.vm.cardText).toBe('stepper.FS.selected')
    })
})
