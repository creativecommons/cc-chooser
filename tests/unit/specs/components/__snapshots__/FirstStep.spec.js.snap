// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstStep Component Mark up is correctly rendered 1`] = `
<div
  class="step-content"
>
  <div
    class="step-actions"
  >
    <div
      class="field not-selected"
    >
      <label
        class="b-radio radio"
      >
        <input
          type="radio"
          value="yes"
        />
         
        <span
          class="check"
        />
         
        <span
          class="control-label"
        >
          <span
            class="vocab-body body-normal"
          >
            
                    stepper.yesstepper.FS.selected
                
          </span>
        </span>
      </label>
    </div>
     
    <div
      class="field selected"
    >
      <label
        class="b-radio radio"
      >
        <input
          type="radio"
          value="no"
        />
         
        <span
          class="check"
        />
         
        <span
          class="control-label"
        >
          <span
            class="vocab-body body-normal"
          >
            
                    stepper.nostepper.FS.not-selected
                
          </span>
        </span>
      </label>
    </div>
  </div>
</div>
`;
